includes/class-freemius.php

```diff
@@ -7407,6 +7407,11 @@ function _activate_plugin_event_hook() {
                     $this->apply_filters( 'deactivate_on_activation', true )
                 ) {
                     deactivate_plugins( $other_version_basename );
+                } else if ( $is_premium_version_activation ) {

```

Comment:
Now thinking about it again, perhaps we would want to also handle "Free plugin activation", for example user activates pro version, then deactivates free version on their own. Then notices the pro doesn't work without the Free, so they activate the Free version again. So instead of the `else` part here, we can do something like (pseudo code)

```php
// If both free and premium version are active, then remove SDK from the Free version to avoid having it load the configuration.
if (is_plugin_active( $other_version_basename ) ) {
  fs_remove_sdk_reference_by_basename( $this->_free_plugin_basename );
}
```

But please double check me.

Reply:
@swashata In that use case, when the premium version is activated, the latest SDK is already set to the SDK that comes with the premium version (when using the code in this PR). When the free version is deactivated and reactivated, it will not have any effect on the latest SDK information. Please let me know if I missed something :) Thanks!